### ISSUE TITLE
Node loading is slow.

### ISSUE THREAD

**Issue by zhangp365** [AlekPet/ComfyUI_Custom_Nodes_AlekPet#34]
Loading this customer node takes more than 10 seconds, perhaps some adjustments to the logic could be considered
![20240123142745](https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/assets/144313702/a99f9bf7-b4f5-4958-b04d-0c824f04fad5)

**Comment by AlekPet** [comment]
The fact is that the argotranslate uses a neural network and it installs all sorts of different things, you can simply delete it if you don’t use it, it just doesn’t save all the dependencies in the python environment.

**Comment by Pythonpa** [comment]
Yes, this node took me about six or seven seconds to load, which is the longest load time of any node I have installed so far.

**Comment by AlekPet** [comment]
This is due to the import of ArgosTranslate specified in __init__.py and insertion into mapping.

**Comment by Maslino** [comment]
> This is due to the import of ArgosTranslate specified in **init**.py and insertion into mapping. https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L231-L234
> 
> https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L250-L251
> 
> https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L266-L267

I don't think importing class and updating dict are time cost operations. In fact, I suspect the slow load is due to the module install procedure when checking modules as the following code shows:

https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/d8629270c944f2328d890d03a6a73e14c428ef75/__init__.py#L79-L95

The module_install function creates one subprocess and two threads which are a little heavy. Maybe the two threads are not necessary if using subprocess.check_call or subprocess.check_output instead of subprocess.Popen

**Comment by AlekPet** [comment]
> > This is due to the import of ArgosTranslate specified in **init**.py and insertion into mapping. https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L231-L234
> > 
> > https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L250-L251
> > 
> > https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L266-L267
> 
> I don't think importing class and updating dict are time cost operations. In fact, I suspect the slow load is due to the module install procedure when checking modules as the following code shows:
> 
> https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/d8629270c944f2328d890d03a6a73e14c428ef75/__init__.py#L79-L95
> 
> The module_install function creates one subprocess and two threads which are a little heavy. Maybe the two threads are not necessary if using subprocess.check_call or subprocess.check_output instead of subprocess.Popen

There is no line from the code I provided, I thought you wanted to turn off ArgosTranslate. Let's see what we can do.

**Comment by Maslino** [comment]
> > > This is due to the import of ArgosTranslate specified in **init**.py and insertion into mapping. https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L231-L234
> > > https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L250-L251
> > > https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/master/__init__.py#L266-L267
> > 
> > 
> > I don't think importing class and updating dict are time cost operations. In fact, I suspect the slow load is due to the module install procedure when checking modules as the following code shows:
> > https://github.com/AlekPet/ComfyUI_Custom_Nodes_AlekPet/blob/d8629270c944f2328d890d03a6a73e14c428ef75/__init__.py#L79-L95
> > 
> > The module_install function creates one subprocess and two threads which are a little heavy. Maybe the two threads are not necessary if using subprocess.check_call or subprocess.check_output instead of subprocess.Popen
> 
> There is no line from the code I provided, I thought you wanted to turn off ArgosTranslate, after delete folder in custom_nodes. Let's see what we can do.

Oh, I misunderstood it. After deleting sub folder, some code in __init__.py should also be deleted. What I mentioned foreward may be another perfomance optimization point.

**Comment by AlekPet** [comment]
Yes, that’s right, you need to remove the folder from the ComfyUi_Custom_Nodes_Alekpet/ArgosTranslate folder and delete or comment out those lines where argos is imported into __init_.py. There is another way - download the old __init__.py and replace the original one, where nodes were added to my mapping automatically and also delete the folder.

**Comment by AlekPet** [comment]
Updated *__init__.py*, added check for already installed modules and other code optimizations
